### ISSUE TITLE
[DOC]

### ISSUE THREAD

**Issue by travisestill** [elastic/elasticsearch#88232]
### Elasticsearch Version

7.11 and later

### Java Version

_bundled_

### OS Version

N/A

### Problem Description

Restoring a snapshot with a feature state requires specific values to be send with the restore command, but we have not documented what these values are in any of the following:

- https://www.elastic.co/guide/en/elasticsearch/reference/current/snapshots-restore-snapshot.html#restore-feature-state
- https://www.elastic.co/guide/en/elasticsearch/reference/current/restore-snapshot-api.html#restore-snapshot-api-feature-states

Please update the docs to include:

- The valid values of feature states.
- The directive to exclude other indices. 
-  API examples. 

### Steps to Reproduce

N/A

**Comment by lucabelluccini** [comment]
Example on how to restore ONLY the `security` feature indices (without cluster state nor additional indices).
Might be useful for final users. Also highlight it is NOT required to close them, which is surprising compared to normal indices.
```
POST _snapshot/found-snapshots/mysecuritysnapshot/_restore
{
  "include_global_state": false,
  "feature_states": ["security"],
  "indices": "-*"
}
```

**Comment by anthonymcglone2022** [comment]
@mayya-sharipova @romain-chanu could this issue be taken by an external contributor? If so, I'd like to work on the updates.

**Comment by mayya-sharipova** [comment]
@anthonymcglone2022 Yes, please, go ahead. Then somebody from our  @elastic/es-docs or @elastic/es-distributed teams will review your work.

**Comment by romain-chanu** [comment]
@anthonymcglone2022 - feel free to contribute, I can also take a look at the changes once they are ready.
